### DISCUSSION TITLE
A Solar Charger energy driver: channel names (aka phase designation)

### DISCUSSION THREAD

**Discussion by meMorizE** [arendst/Tasmota#21402] [General]
Hi,
I am finalising an energy driver for a MakeSkyBlue Solar Charge Controller (http://www.makeskyblue.com):

![Screenshot 2024-05-11 at 23-02-47 Tasmota - Main Menu](https://github.com/arendst/Tasmota/assets/7398631/82bf9de8-c90f-4fdd-a52e-b62ee128505f)

The phase notations L1 and L2 does not make any sense here.
L1 shows the DC measurements related to the solar input.
L2 shows the DC measurements related to the battery.

I understand, that energy drivers have their AC single/multi phase roots, but these days DC energy is getting more common, especially around solar charging.

### 'Half' fix: Suppress by existing information

The tEnergy structure (xdrv_03_energy.ino and xdrv_03_esp32_energy.ino) has already a dc flag:
```
...
bool type_dc;
...
```

It is used rarely but I suggest to use this flag to also suppress Lx phase headings.

### Full fix: Enhancement with pointer to channel strings

A new structure member will be added at tEnergy:
```
...
char * channel_name[ENERGY_MAX_PHASES]
...
```

Initialized with nullpointer to keep current behavior.
Every energy driver can set it up with the desired channel designators.

Any comments ?

**Comment by alexeytm82** [comment]
Hello. I didn't find this device in the supported ones. Can you share your experiences?

**Reply by meMorizE** [reply]
It's some time ago, but it is still in use on my side...

I have created and used it to flash the original V119 Wifi Box hardware which includes an ESP8285 with 1MB Flash.
The orignal firmware was isolated to an android app usage...

It uses a serial UART connection to the solar charger (9600 bit/s, 5V), which uses a simple request-response protocol.
Sometimes the solar charger does not respond, but this is seldom (mean lost is about 0.2%)

I want to upgrade to a ESP32 hardware soon, I will restart working on it and maybe provide a refresh and PR the next days.
I have already seen that xnrg_25 is now used by another driver, so I have to increment some parts.

**Comment by Jason2866** [comment]
@meMorizE  A PR is welcome which introduces no breaking changes and allows the configuration wanted.

**Reply by meMorizE** [reply]
Got it, after a successful PR-process of the driver we can further discuss the channel name issue of this thread.

**Reply by meMorizE** [reply]
This is an actual snapshot:
<img width="576" height="1073" alt="Screenshot 2025-10-25 133030" src="https://github.com/user-attachments/assets/aa8f52d1-ae81-4f0c-b26b-554cca61f1b7" />

Since my old development the 'Active Power Total' is new, so I have now forced battery power to 0, otherwise it get added at total.

My general question:
Is it useful to build the battery part as L2 (to see voltage and current, but power not anymore) or does it make more sense to only implement L1 (solar) and add variables of battery voltage current and maybe power separately (like the temperature) ?

**Reply by meMorizE** [reply]
Question selfanswered (finally making the topic obsolete):

Only the Solar voltage, current and power will be represented as one phase. This gives more opportunities, which are currently under construction. The Battery voltage, charge current, temperature and some more will be additional informations, posted as columns after the energy data.

The L1 L2 L3... designation will match to the related Serial-Interface module configuration of the Multi-Serial support.

**Comment by transmitter3000** [comment]
Im taking a brake for a few days and that is fine

**Comment by meMorizE** [comment]
Finally I want to close this topic, because a phase name designation is not useful for this energy driver.

In the meanwhile there is a PR of the energy driver:
https://github.com/arendst/Tasmota/pull/24151
